analysis/parser/RawTextParser/src/main/kotlin/de/maibornwolff/codecharta/parser/rawtextparser/ProjectGenerator.kt

```diff
@@ -5,31 +5,33 @@ import de.maibornwolff.codecharta.model.MutableNode
 import de.maibornwolff.codecharta.model.PathFactory
 import de.maibornwolff.codecharta.model.Project
 import de.maibornwolff.codecharta.model.ProjectBuilder
-import de.maibornwolff.codecharta.parser.rawtextparser.model.FileMetrics
 
-class ProjectGenerator() {
-    private lateinit var projectBuilder: ProjectBuilder
+class ProjectGenerator(private val projectBuilder: ProjectBuilder = ProjectBuilder()) {
 
-    fun generate(metricMap: Map<String, FileMetrics>, pipedProject: Project?): Project {
-        projectBuilder = ProjectBuilder()
-        metricMap.forEach { addAsNode(it) }
+    fun generate(projectMetrics: ProjectMetrics, pipedProject: Project?): Project {
+        addMetricsAsNodes(projectMetrics.metricsMap)
         var project = projectBuilder.build()
+
         if (pipedProject != null) {
             project = MergeFilter.mergePipedWithCurrentProject(pipedProject, project)
         }
+
         return project
     }
 
-    private fun addAsNode(metrics: Map.Entry<String, FileMetrics>) {
-        var directory = ""
-        var fileName = metrics.key
-        if (fileName.contains("/")) {
-            directory = metrics.key.substringBeforeLast("/")
-            fileName = metrics.key.substringAfterLast("/")
+    private fun addMetricsAsNodes(metricsMap: Map<String, FileMetrics>) {
+        metricsMap.forEach { (key, fileMetrics) ->
+            val (directory, fileName) = splitDirectoryAndFileName(key)
+            val node = MutableNode(fileName, attributes = fileMetrics.metricsMap)
+            val path = PathFactory.fromFileSystemPath(directory)
+            projectBuilder.insertByPath(path, node)
         }
+    }
 
-        val node = MutableNode(fileName, attributes = metrics.value.metricMap)
-        val path = PathFactory.fromFileSystemPath(directory)
-        projectBuilder.insertByPath(path, node)
+    private fun splitDirectoryAndFileName(path: String): Pair<String, String> {
+        val lastSlashIndex = path.lastIndexOf("/")
```

Comment:
Can we rely on linux formatted pathes?
There are other path separators like backslash.
Here are some java examples, on how to deal with [this](https://www.baeldung.com/java-filename-full-path).